env0/resource_agent_project_assignment_test.go

```diff
@@ -11,219 +11,215 @@ import (
-						"agent_id":   agentId,
-					}),
-					Check: resource.ComposeAggregateTestCheckFunc(
-						resource.TestCheckResourceAttr(accessor, "id", agentId+"_"+projectId),
-						resource.TestCheckResourceAttr(accessor, "project_id", projectId),
-						resource.TestCheckResourceAttr(accessor, "agent_id", agentId),
-					),
+					Config: getConfig(&assignment),
+					Check:  getCheck(&assignment),
 				},
 				{
-					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
-						"project_id": projectId,
-						"agent_id":   agentId,
-					}),
+					Config: getConfig(&updatedAssignment),
+					Check:  getCheck(&updatedAssignment),
 				},
 			},
 		}
 
 		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
 			gomock.InOrder(
-				mock.EXPECT().ProjectsAgentsAssignments().Times(1).Return(GenerateProjectsAgentsAssignments("p111", "a222"), nil),
-				mock.EXPECT().AssignAgentsToProjects(GenerateProjectsAgentsAssignmentsMap(
-					"p111",
-					"a222",
-					projectId,
-					agentId,
-				)).Times(1).Return(nil, nil),
-				mock.EXPECT().ProjectsAgentsAssignments().Times(4).Return(GenerateProjectsAgentsAssignments("p111", "a222", projectId, agentId), nil),
-				mock.EXPECT().AssignAgentsToProjects(GenerateProjectsAgentsAssignmentsMap(
-					"p111",
-					"a222",
-				)).Times(1).Return(nil, nil),
+				mock.EXPECT().AssignAgentsToProjects(getAssignmentPayload(&assignment)).Times(1).Return(nil, nil),
+				mock.EXPECT().ProjectsAgentsAssignments().Times(2).Return(getAssignmentsResponse([]AgentProjectAssignment{otherAssignment, assignment}), nil),
+				mock.EXPECT().AssignAgentsToProjects(getAssignmentPayload(&updatedAssignment)).Times(1).Return(nil, nil),
+				mock.EXPECT().ProjectsAgentsAssignments().Times(1).Return(getAssignmentsResponse([]AgentProjectAssignment{otherAssignment, updatedAssignment}), nil),
+				mock.EXPECT().AssignAgentsToProjects(getAssignmentPayload(&deleteAssignment)).Times(1).Return(nil, nil),
 			)
 		})
 	})
 
-	t.Run("Create assignment with Drift", func(t *testing.T) {
+	t.Run("default agent when assignment not found - validate no drift", func(t *testing.T) {
 		testCase := resource.TestCase{
 			Steps: []resource.TestStep{
 				{
-					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
-						"project_id": projectId,
-						"agent_id":   agentId,
-					}),
-					ExpectNonEmptyPlan: true,
-				},
-				{
-					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
-						"project_id": projectId,
-						"agent_id":   agentId,
-					}),
-					Check: resource.ComposeAggregateTestCheckFunc(
-						resource.TestCheckResourceAttr(accessor, "id", agentId+"_"+projectId),
-						resource.TestCheckResourceAttr(accessor, "project_id", projectId),
-						resource.TestCheckResourceAttr(accessor, "agent_id", agentId),
-					),
-					PlanOnly:           true,
-					ExpectNonEmptyPlan: true,
+					Config: getConfig(&defaultAssignment),
```

Comment:
I don't think that `ENV0_DEFAULT` is an actual agentId that is returned from API ever. I think it's just a value that is accepted when updating assignments, to indicate that this is the default agent

Reply:
Yeah I got that. that's what I did. Or am I misunderstanding?

Reply:
Not sure why I commented that here 😅